src/libzstd.jl

```diff
@@ -89,19 +87,21 @@ mutable struct DStream
 end
 
 function initialize!(dstream::DStream)
-    return ccall((:ZSTD_initDStream, libzstd), Csize_t, (Ptr{Cvoid},), dstream.ptr)
+    return LibZstd.ZSTD_initDStream(dstream.ptr)
```

Comment:
Do NOT do these object to pointer conversions manually. Leave it to ccall to do it. I’m assuming clang isn’t generating a signature that requires you to do such conversion manually. If that’s not the case then the code it generate is broken and must be fixed. ( a brief look suggests that the generated wrappers are correct)

same with all uses of pointer_from_objref here.

Reply:
Actually I just realized that the alignment made me think that the change is new to this pr but it’s actually not….


I that case it should still be fixed but doesn’t have to be done here.

Reply:
I'm unclear what you are suggesting here.

`LibZstd.ZSTD_initDStream` now calls `ccall`. All I'm doing with the change above is consolidating all the `ccall` references into the `LibZstd` module.
https://github.com/JuliaIO/CodecZstd.jl/blob/c084472af8cf40d30cab56001f866d4700ac0e31/src/libzstd/LibZstd.jl#L307-L309

Perhaps you mean that I should just pass `dstream` directly rather than `dstream.ptr` or something like that?

Reply:
Yes the change is fine. But both the new and the old version should have the Julia object (dsteam in this case) directly ccall. You’ll need to define the correct unsafe convert to make it work.

Reply:
I see what you mean. This probably has some garbage collection implications as well now that I think about it.

I think I might want to save those changes for another PR and perhaps another release.

Reply:
The only reason it’s broken is indeed because of object lifetime.